trajopt_ifopt/include/trajopt_ifopt/constraints/collision_evaluators.h

```diff
@@ -0,0 +1,459 @@
+  Eigen::Isometry3d world_to_base_;
+  TrajOptCollisionConfig collision_config_;
+  tesseract_environment::StateSolver::Ptr state_solver_;
+  CollisionExpressionEvaluatorType evaluator_type_;
+  std::function<tesseract_environment::EnvState::Ptr(const std::vector<std::string>& joint_names,
+                                                     const Eigen::Ref<const Eigen::VectorXd>& joint_values)>
+      get_state_fn_;
+  bool dynamic_environment_;
+
+  /**
+   * @brief This takes contacts results at each interpolated timestep and creates a single contact results map.
+   * This also updates the cc_time and cc_type for the contact results
+   * @param contacts_vector Contact results map at each interpolated timestep
+   * @param contact_results The merged contact results map
+   * @param dt delta time
+   */
+  void processInterpolatedCollisionResults(std::vector<tesseract_collision::ContactResultMap>& contacts_vector,
+                                           tesseract_collision::ContactResultMap& contact_results,
+                                           double dt) const;
+
+  /**
+   * @brief Remove any results that are invalid.
+   * Invalid state are contacts that occur at fixed states or have distances outside the threshold.
+   * @param contact_results Contact results vector to process.
+   */
+  void removeInvalidContactResults(tesseract_collision::ContactResultVector& contact_results,
+                                   const Eigen::Vector2d& pair_data) const;
+
+private:
+  CollisionEvaluator() = default;
+};
+
+/**
+ * @brief This collision evaluator only operates on a single state in the trajectory and does not check for collisions
+ * between states.
+ */
+struct DiscreteCollisionEvaluator : public CollisionEvaluator
+{
+public:
+  using Ptr = std::shared_ptr<DiscreteCollisionEvaluator>;
+  using ConstPtr = std::shared_ptr<const DiscreteCollisionEvaluator>;
+
+  DiscreteCollisionEvaluator(tesseract_kinematics::ForwardKinematics::ConstPtr manip,
+                             tesseract_environment::Environment::ConstPtr env,
+                             tesseract_environment::AdjacencyMap::ConstPtr adjacency_map,
+                             const Eigen::Isometry3d& world_to_base,
+                             const TrajOptCollisionConfig& collision_config,
+                             bool dynamic_environment = false);
+
+  void CalcCollisions(const std::vector<double>& x, tesseract_collision::ContactResultMap& dist_results) override;
+  /**
+   * @brief Given joint names and values calculate the collision results for this evaluator
+   * @param dof_vals Joint values set prior to collision checking
+   * @param dist_results Contact Results Map
+   */
+  void CalcCollisions(const Eigen::Ref<const Eigen::VectorXd>& dof_vals,
+                      tesseract_collision::ContactResultMap& dist_results);
+
+private:
+  tesseract_collision::DiscreteContactManager::Ptr contact_manager_;
+};
+
+///**
```

Comment:
Should the comment code below be removed?

Reply:
The commented code is what is left to be implemented and will be kept until then.